src/audio/component.c

```diff
@@ -29,31 +29,45 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
-	uint32_t flags;
-
-	irq_local_disable(flags);
 
-	/* search driver list for driver type */
-	list_for_item(clist, &drivers->list) {
-		info = container_of(clist, struct comp_driver_info, list);
-		if (info->drv->type == type) {
-			drv = info->drv;
-			platform_shared_commit(info, sizeof(*info));
-			goto out;
+	/* use UUID to match the driver if UUID is provided */
+	if (sof_is_uuid_nil(comp->uuid)) {
+		/* search driver list with UUID */
+		list_for_item(clist, &drivers->list) {
+			info = container_of(clist, struct comp_driver_info,
+					    list);
+			if (!memcmp(info->drv->uid, comp->uuid, UUID_SIZE)) {
+				tr_dbg(&comp_tr,
+				       "get_drv_from_uuid(), found driver type %d, uuid %s",
+				       info->drv->type,
+				       info->drv->tctx->uuid_p);
+				drv = info->drv;
+				break;
+			}
+		}
+		tr_err(&comp_tr, "get_drv(): the provided UUID (%8x%8x%8x%8x) doesn't match to any driver!",
+		       *(uint32_t *)&comp->uuid[0],
+		       *(uint32_t *)&comp->uuid[4],
+		       *(uint32_t *)&comp->uuid[8],
+		       *(uint32_t *)&comp->uuid[12]);
+	} else {
+		/* search driver list for driver type */
+		list_for_item(clist, &drivers->list) {
+			info = container_of(clist, struct comp_driver_info,
+					    list);
+			if (info->drv->type == comp->type) {
```

Comment:
I don't understand this part either, now you've removed the interrupt handling and platform_shared_commit in all cases, is this intentional? If yes, you want to remove these two parts first in a separate patch before merging the two cases.

Reply:
It is intentional, let me add a separate commit to remove them ahead.

BTW, @plbossart as a heads up, please be noticed that I am working on a new solution (proposed by @lgirdwood 6 hours back) to remove any potential ABI backward compatible issues introduced by this UUID change, with putting the UUID array to append to the whole component structure, and use the sof_ipc_comp.reserved member as its offset to the component struct. So this PR here will need the corresponding change. I will add explanation about it to the sof-docs PR tomorrow.